main.go

```diff
@@ -76,13 +76,23 @@ func (s *serialPort) Write(b []byte) (int, error) {
 
 var deferredFuncs []func()
 
+// RegisterAgent is a blocking call, the RPC itself has retry logic baked it
+// with jitter and backoff up to a total of 10 minutes.
+// If client creation or register agent (after retries) fail we then wait for
+// 5 minutes and try again.
 func registerAgent(ctx context.Context) {
-	if agentconfig.TaskNotificationEnabled() || agentconfig.GuestPoliciesEnabled() {
-		if client, err := agentendpoint.NewClient(ctx); err != nil {
-			logger.Errorf(err.Error())
-		} else if err := client.RegisterAgent(ctx); err != nil {
-			logger.Errorf(err.Error())
+	for {
+		if agentconfig.TaskNotificationEnabled() || agentconfig.GuestPoliciesEnabled() {
+			if client, err := agentendpoint.NewClient(ctx); err != nil {
+				logger.Errorf(err.Error())
+			} else if err := client.RegisterAgent(ctx); err != nil {
+				logger.Errorf(err.Error())
+			} else {
+				// RegisterAgent completed successfully.
```

Comment:
Would it be helpful to add a log statement here so that the lack of it will be a signal that registeragent was not successful?

Reply:
The agent logs the failures and all the retries as errors, there really isn't a need to log the success case as it's not helpful to customers. When debug logging is enabled it will log the RPC request/response